BITCORNService/Utils/DbActions/DbOperations.cs

```diff
@@ -43,27 +44,25 @@ public static async Task<bool> IsBlockchainTransactionLogged(this BitcornContext
             return await dbContext.CornTx.AnyAsync(w => w.BlockchainTxId == txId);
         }
 
-        public static async Task SaveAsync(this BitcornContext dbContext)
+        public static async Task SaveAsync(this BitcornContext dbContext, IDbContextTransaction transaction)
         {
 
             //create execution strategy so the request can retry if it fails to connect to the database
             var strategy = dbContext.Database.CreateExecutionStrategy();
             await strategy.ExecuteAsync(async () =>
             {
-                using (var transaction = dbContext.Database.BeginTransaction(IsolationLevel.RepeatableRead))
+                try
                 {
-                    try
-                    {
-                        await dbContext.SaveChangesAsync();
-                        transaction.Commit();

```

Comment:
should this come before the save?

Reply:
db transactions always are committed after save